### ISSUE TITLE
Mismatch of loss shape in document with output of TransfoXL

### ISSUE THREAD

**Issue by TobiasLee** [huggingface/transformers#3615]
# 🐛 Bug

## Information

Model: Transformer-XL

Language: English

The problem arises when using: a small demo of transformer-xl output

## To reproduce

```python3
model_name = 'transfo-xl-wt103'
model = AutoModelWithLMHead.from_pretrained(model_name)
tokenizer = AutoTokenizer.from_pretrained(model_name)
sentence = 'Hello world'
tokenize_input = tokenizer.tokenize(sentence) 
tensor_input = torch.tensor([tokenizer.convert_tokens_to_ids(tokenize_input)])
outputs = model(tensor_input, labels=tensor_input)
print(outputs[0].size())
```
run the code above 

## Expected behavior

The  output[0] is supposed to be a  tensor shaped as `(1,)`, as described in the [source code](https://github.com/huggingface/transformers/blob/master/src/transformers/modeling_transfo_xl.py#L862), however, the actual shape is (1, 2)  (bsz, sequence_len).

## Environment info

     
- `transformers` version: 2.7.0
- Platform: Ubuntu 18.04
- Python version: 3.6
- PyTorch version (GPU?): 1.4.0, with 1080Ti 
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

**Comment by patrickvonplaten** [comment]
The output shape looks correct to me 
"loss (:obj:`torch.FloatTensor` of shape `(batch_size, sequence_length)`" means that 
`outputs[0]` should be of shape ` [batch_size, sequence_length] ` and it is ` [1, 2] `, so that is correct no?

**Comment by TobiasLee** [comment]
yes, since the document was updated via #3661
